master/buildbot/monkeypatches/__init__.py

```diff
@@ -118,6 +118,17 @@ def patch_unittest_testcase():
     if not getattr(TestCase, "assertRegex", None):
         TestCase.assertRegex = TestCase.assertRegexpMatches
 
+@onlyOnce
+def patch_testcase_flushLoggedErrors():
+    from twisted.trial.unittest import SynchronousTestCase
+    import gc
+
+    # DebugInfo use of __del__ leads to hard to find gc timing issues
+    # https://twistedmatrix.com/trac/ticket/9506#ticket
```

Comment:
This was closed as a duplicate of 9505. Update to point to that one instead.

Reply:
A comment was left on 9505 that the test needs to be rewritten to not drop the Deferred.

Reply:
```
git grep -E '^ +self.assertFailure' |wc -l
26
```
:-(

Reply:
https://github.com/buildbot/buildbot/issues/4253